packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -49,32 +49,54 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         return name.replace("@", "").replace("/", "-");
     }
 
+    /**
+     * Take an unknown error object and add the appropriate info from it to the event
+     * NOTE - message and stack will be copied over from the error object,
+     * along with other telemetry properties if it's an ILoggingError
+     * @param event - Event being logged
+     * @param error - Error to extract info from
+     * @param fetchStack - Whether to fetch the current callstack if error.stack is undefined
+     */
     public static prepareErrorObject(event: ITelemetryBaseEvent, error: any, fetchStack: boolean) {
-        if (error === null || typeof error !== "object") {
+        if (typeof error !== "object" || error === null) {
             event.error = error;
         } else {
-            // WARNING: Exceptions can contain PII!
-            // For example, XHR will throw object derived from Error that contains config information
-            // for failed request, including all the headers, and thus - user tokens!
-            // Extract only call stack, message, and couple network-related properties form error object
-
-            const errorAsObject = error as {
-                stack?: string;
-                message?: string;
-            };
-
+            const errorAsObject = error as Partial<Error>;
             event.stack = errorAsObject.stack;
             event.error = errorAsObject.message;
 
-            // Error message can contain PII information.
-            // If we know for sure it does, we have to not log it.
-            if (error.containsPII) {
-                event.error = "Error message was removed as it contained PII";
-            } else if (error.getTelemetryProperties) {
-                const telemetryProps: ITelemetryProperties = error.getTelemetryProperties();
-                for (const key of Object.keys(telemetryProps)) {
-                    if (event[key] === undefined) {
-                        event[key] = telemetryProps[key];
+            if (isILoggingError(error)) {
+                const taggableProps = error.getTelemetryProperties();
+                for (const key of Object.keys(taggableProps)) {
+                    if (event[key] !== undefined) {
+                        continue; //* test this
+                    }
+                    const taggableProp = taggableProps[key];
+                    const { value, tag } = (typeof taggableProp === "object")
+                        ? taggableProp
+                        : { value: taggableProp, tag: TelemetryDataTag.None };
+                    switch (tag) {
+                        case TelemetryDataTag.None:
+                        case TelemetryDataTag.PackageData:
+                            // For Microsoft applications, PackageData is safe for now
```

Comment:
It's very easy in the future to read it that treatment of None & PackageData should stay the same and possibly not send data. I think that's not the intention - I'd duplicate that code for None case (without comment).